File: 121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int lowest = INT_MAX;
        int maxProfit = 0;
        for (int p : prices) {
            if (p <= lowest) {
                lowest = p;
            } else {
                maxProfit = max(maxProfit, p - lowest);
            }
        }
        return maxProfit;
    }
};

File: java/121. Best Time to Buy and Sell Stock.java
// O(n)
public class Solution {
    public int maxProfit(int[] prices) {
        int result = 0;
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] > min) {
                result = Math.max(result, prices[i] - min);
            } else {
                min = prices[i];
            }
        }
        return result;
    }
}
